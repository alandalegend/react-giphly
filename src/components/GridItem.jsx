import { useState } from "react";
import PropTypes from "prop-types";

export const GifItem = ({ title, url, id }) => {
  const [inputValue, setInputValue] = useState([]);

  return (
    <div className="card">
      <img src={url} alt={title}/>
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
