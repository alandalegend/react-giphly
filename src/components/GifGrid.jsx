import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GridItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h4 className="">Cargando...</h4>}
      <div className="card-grid">
        {images.map((image) => (
          //   <li key={image.id}>{image.title}</li>
          <GifItem key={image.id} {...image} />
        ))}
      </div>
      {/* </ol> */}
    </>
  );
};
