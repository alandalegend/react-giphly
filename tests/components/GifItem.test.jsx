// import { fireEvent, render, screen } from "@testing-library/react";

import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GridItem";

describe("Pruebas en <GifItem />", () => {
  const title = "2";
  const url = "http://dalegend.net/foto.jpg";

  test("debe de hhacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    //Creo mi snapshot
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
     expect(screen.getByText(title)).toBeTruthy;

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
});
