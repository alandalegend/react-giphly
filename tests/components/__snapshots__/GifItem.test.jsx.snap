// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hhacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="2"
      src="http://dalegend.net/foto.jpg"
    />
    <p>
      2
    </p>
  </div>
</div>
`;
